Audacia.Quest/Audacia.Quest/Pages/Index.razor

```diff
@@ -1,70 +1,127 @@
 ﻿@page "/"
 
 @using System.Drawing
+@using Audacia.Quest.Core
+@using Audacia.Quest.Core.Events
+@using Audacia.Quest.Core.Params
 @using Blazor.Extensions
 @using Blazor.Extensions.Canvas.Canvas2D
 
 @inject IJSRuntime JsRuntime
 
-<div id="canvasContainer" style="position: fixed; opacity: 1; background-color: #6495ED; width: 100%; height: 100%">
+@*<img @ref="_fox" hidden src="assets/fox.png" />*@
+@if (Initialized)
+{
+    <Assets Context="_gameContext" />
+}
+
+<div id="canvasContainer" style="background-color: #6495ED; position: fixed; opacity: 1; width: 100%; height: 100%">
     <BECanvas @ref="_canvas"></BECanvas>
 </div>
 
 @code
 {
     private Canvas2DContext _context;
-    protected BECanvasComponent _canvas;
+    private BECanvasComponent _canvas;
+    private Game _game;
+    private BlazorGameContext _gameContext;
 
-    bool _initialized = false;
+    [Parameter]
+    public bool Initialized { get; set; } = false;
 
     protected override async Task OnAfterRenderAsync(bool firstRender)
     {
-        if (!_initialized)
+        _context = await this._canvas.CreateCanvas2DAsync();
+
+        if (!Initialized)
         {
             await InitGame();
-            _initialized = true;
+            Initialized = true;
+            this.StateHasChanged();
         }
-
-        this._context = await this._canvas.CreateCanvas2DAsync();
-
-        //await this._context.ClearColorAsync(0.39f, 0.58f, 0.92f, 1);
-        //await this._context.ClearAsync(BufferBits.COLOR_BUFFER_BIT);
     }
 
     private async ValueTask InitGame()
     {
         var canvasMeta = await JsRuntime.InvokeAsync<CanvasMeta>("initCanvas", DotNetObjectReference.Create(this));
-
         _canvas.Height = canvasMeta.Height;
         _canvas.Width = canvasMeta.Width;
 
-        // this will make sure that the viewport is correctly initialized
         await JsRuntime.InvokeAsync<object>("gameWindowResize", DotNetObjectReference.Create(this));
+
+        EventsCollection.Add(EventConstants.RESIZE, new ResizeEvent());
```

Comment:
I would suggest to move these into a new method, just for the event that we decide to add "asdw" will make this `Init` method more structured